source/simulator/solver_schemes.cc

```diff
@@ -516,12 +515,26 @@ namespace aspect
 
         assemble_newton_stokes_system = assemble_newton_stokes_matrix = true;
 
-        denormalize_pressure (last_pressure_normalization_adjustment,
-                              linearized_stokes_initial_guess,
-                              current_linearization_point);
+        if (nonlinear_iteration == 0)
+          {
+            assemble_newton_stokes_system = assemble_newton_stokes_matrix = false;
+            assemblers.reset (new internal::Assembly::AssemblerLists<dim>());
+            set_assemblers();
+          }
+        else
+          {
+            denormalize_pressure (last_pressure_normalization_adjustment,
+                                  linearized_stokes_initial_guess,
+                                  current_linearization_point);
+          }
 
         if (nonlinear_iteration <= 1)
-          compute_current_constraints ();
+          {
+            // So this should only happen on nonlinear iteration 0 and 1.
```

Comment:
That comment is not very useful as it just restates the if above.

I also don't understand why you need this block because you do the same thing in line 521 above if nonlinear_iteration == 0

Reply:
good point. I will remove the lines 
``` 
assemblers.reset (new internal::Assembly::AssemblerLists<dim>());
set_assemblers();
```
at `if (nonlinear_iteration == 0)`, and the unnecessary comment.